pkg/webhook/kapiserver/ensurer.go

```diff
@@ -87,8 +89,13 @@ func NewEnsurer(mgr manager.Manager, logger logr.Logger) genericmutator.Ensurer
 // ensureKubeAPIServerIsMutated ensures that the kube-apiserver deployment is mutated accordingly
 // so that it is able to communicate with the oidc-webhook-authenticator
 func ensureKubeAPIServerIsMutated(ps *corev1.PodSpec, c *corev1.Container, caBundleSecretName string) {
-	c.Command = extensionswebhook.EnsureStringWithPrefix(c.Command, oidcWebhookConfigPrefix, constants.AuthenticatorDir+"/kubeconfig")
-	c.Command = extensionswebhook.EnsureStringWithPrefix(c.Command, oidcWebhookCacheTTLPrefix, "0")
+	c.Args = extensionswebhook.EnsureStringWithPrefix(c.Args, oidcWebhookConfigPrefix, constants.AuthenticatorDir+"/kubeconfig")
+	c.Args = extensionswebhook.EnsureStringWithPrefix(c.Args, oidcWebhookCacheTTLPrefix, "0")
+
+	// TODO(vpnachev): Remove clean-up of `--authentication-token-webhook-` flags in `command` after version v0.32.0 has been released.
+	c.Command = slices.DeleteFunc(c.Command, func(x string) bool {
+		return strings.HasPrefix(x, oidcWebhookConfigPrefix) || strings.HasPrefix(x, oidcWebhookCacheTTLPrefix)
+	})
```

Comment:
Do we need this cleanup? I am not sure when this will actually remove the flags if we do not set them in the first place.

Reply:
I think I answered my question. You have in mind that the deployment modified manually which will trigger the webhook but not remove the old flags?

Reply:
For example https://github.com/gardener/gardener-extension-shoot-oidc-service/blob/c55f07d30a58f0cacf60cf407e36825b1b1a0bce/pkg/controller/lifecycle/actuator.go#L215 just sends empty patch which triggers the webhook, in this case the kube-apiserver could have been already mutated and the `--authentication-token-webhook-*` flags are set in the `Command`, we would duplicate them in the `Args` if we do not clean them in the `Command`. Honestly, I have not checked but I guess `kube-apiserver` is clever enough to handle duplicated flags, but I prefer to clean those already set in `Command`.